### PR TITLE
Add inputs_by option to invoke_workflow

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

File: bioblend/galaxy/workflows/__init__.py [hunk 1]
```diff
@@ -373,7 +373,10 @@ def invoke_workflow(self, workflow_id, inputs=None, params=None,
 
                        This map may also be indexed by the UUIDs of the workflow steps,
                        as indicated by the ``uuid`` property of steps returned from the
-                       Galaxy API.
+                       Galaxy API. Alternatively workflow steps may be addressed by
+                       the label that can be set in the workflow editor. If using
+                       uuid or label you need to also set the ``inputs_by`` parameter
+                       to ``uuid`` or ``name``.
```
Comment (nsoranzo) [inline bioblend/galaxy/workflows/__init__.py]:
```suggestion
                       to ``step_uuid`` or ``name``.
```

**Comment by nsoranzo** [comment]
Thanks @mvdbeek ! Should we add a test for this to `bioblend/_tests/TestGalaxyWorkflows.py` ?

**Comment by mvdbeek** [comment]
Sure, I could do that.

**Comment by nsoranzo** [comment]
The added test doesn't seem to work unfortunately.

**Comment by mvdbeek** [comment]
yup, though bioblend.ConnectionError: Unexpected HTTP status code: 400: {"err_msg": "Workflow cannot be run because an expected input step '65' has no input dataset.", "err_code": 0} is weird, there shouldn't be a step `'65'`

**Comment by mvdbeek** [comment]
Oh, I guess I looked at the wrong workflow, it seems to work now that I updated the test to the correct labels.

File: bioblend/_tests/TestGalaxyWorkflows.py [hunk 2]
```diff
@@ -167,6 +167,30 @@ def test_run_workflow(self):
...
+        fn = test_util.get_abspath("test_util.py")
+        file_name = "test1"
+        tool_output = self.gi.tools.upload_file(
+            fn,
+            history_id,
+            # First param could be a regular path also of course...
+            file_name=file_name,
+            dbkey="?",
+            file_type="tabular",
+        )
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyWorkflows.py]: I think you can replace lines 175-184 with:
```python
        dataset1_id = self._test_dataset(history_id)
```

File: bioblend/_tests/TestGalaxyWorkflows.py [hunk 3]
```diff
@@ -167,6 +167,30 @@ def test_run_workflow(self):
         with self.assertRaises(Exception):
             self.gi.workflows.run_workflow(wf['id'], None)
 
+    def test_invoke_workflow(self):
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyWorkflows.py]: Can you please add `@test_util.skip_unless_galaxy('release_15.03')` above this line?

File: bioblend/_tests/TestGalaxyWorkflows.py [hunk 4]
```diff
@@ -167,6 +167,30 @@ def test_run_workflow(self):
...
+        file_name = "test1"
+        tool_output = self.gi.tools.upload_file(
+            fn,
+            history_id,
+            # First param could be a regular path also of course...
+            file_name=file_name,
+            dbkey="?",
+            file_type="tabular",
+        )
+        dataset = {'src': 'hda', 'id': tool_output['outputs'][0]['id']}
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyWorkflows.py]:
```suggestion
        dataset = {'src': 'hda', 'id': dataset1_id}
```

**Pull request merged**